sparkmagic/sparkmagic/tests/test_configuration.py

```diff
@@ -117,3 +117,47 @@ def test_share_config_between_pyspark_and_pyspark3():
         conf.base64_kernel_python3_credentials()
         == conf.base64_kernel_python_credentials()
     )
+
+
+def test_get_session_properties():
+    assert conf.get_session_properties("python") == {"kind": "pyspark"}
+
+
+def test_get_session_properties_no_defaults():
+    conf.override(
+        conf.session_configs.__name__, {"foo": "bar", "config": {"foo": "bar"}}
+    )
+    assert conf.get_session_properties("python") == {
+        "kind": "pyspark",
+        "foo": "bar",
+        "config": {"foo": "bar"},
+    }
+
+
+def test_get_session_properties_with_defaults():
+    conf.override(
+        conf.session_configs_defaults.__name__,
+        {"foo": "default", "config": {"foo": "default", "default": "default"}},
+    )
+    assert conf.get_session_properties("python") == {
+        "kind": "pyspark",
+        "foo": "default",
+        "config": {"foo": "default", "default": "default"},
+    }
+
+
```

Comment:
Can you add an explicit test for the recursive merging? Something that is nested at least two levels deep

Reply:
Added more items to the merge test